File: APM/CSharp/Interpreter/Interpreter/repository/Repository.cs
using Interpreter.domain;

namespace Interpreter.repository
{
    public class Repository : IRepository
    {
        private int numberOfPrograms;
        private readonly ProgramState[] programs;

        public Repository()
        {
            programs = new ProgramState[10];
            numberOfPrograms = 0;
        }

        public void Add(ProgramState state)
        {
            programs[numberOfPrograms] = state;
            numberOfPrograms++;
        }


        public ProgramState GetCurentProgramState(int pos)
        {
            if (pos < numberOfPrograms)
            {
                return programs[pos];
            }
            return null;
        }
    }
}

File: APM/Java/Interpreter/src/ro/sandorrobertk94/repository/Repository.java
package ro.sandorrobertk94.repository;

import ro.sandorrobertk94.domain.ProgramState;

/**
 * Created by robert on 10/25/15.
 */
public class Repository implements IRepository {
    private ProgramState[] programs;
    private int numberOfPrograms;

    public Repository() {
        programs = new ProgramState[10];
        numberOfPrograms = 0;
    }

    public void add(ProgramState prgState) {
        programs[numberOfPrograms] = prgState;
        numberOfPrograms++;
    }

    @Override
    public ProgramState getProgState(int pos) {
        if (pos < numberOfPrograms) {
            return programs[pos];
        }
        return null;
    }
}
